src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -503,6 +503,9 @@ abstract class AbstractCheckTask(
 
     protected fun ProtoToIMessageConverter.fromProtoPreFilter(protoPreMessageFilter: RootMessageFilter,
                                                               messageName: String = protoPreMessageFilter.messageType): IMessage {
+        check(messageName.isNotBlank()) {
+            "Rule cannot be created or executed because the root message filter does not contain 'message type'"
+        }
```

Comment:
I think it is a bit late to check the message type. I think we should do that in the RuleFactory before rule creation. That will allow us to report an error before submitting the rule